File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int sum = 0; 
        int maxi = INT_MIN; // Use INT_MIN instead of LONG_MIN

        for(int i = 0; i < n; i++) {
            sum += nums[i];
            maxi = max(maxi, sum);
            if(sum < 0) {
                sum = 0;
            }
        }

        return maxi;
       
    }
};

File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n = len(nums)
        curr_sum = 0
        maxi = -sys.maxsize 

        for i in range(n):
            curr_sum += nums[i]
            maxi = max(maxi, curr_sum)
            if curr_sum < 0:
                curr_sum = 0
        return maxi